spec/unit/data_collector/resource_report_spec.rb

```diff
@@ -0,0 +1,146 @@
+
+  describe "#skipped" do
+    let(:conditional) { double("Chef::Resource::Conditional") }
+
+    it "should set status and conditional" do
+      report.skipped(conditional)
+      expect(report.conditional).to eq conditional
+      expect(report.status).to eq "skipped"
+    end
+  end
+
+  describe "#up_to_date" do
+    it "should set status" do
+      report.up_to_date
+      expect(report.status).to eq "up-to-date"
+    end
+  end
+
+  describe "#updated" do
+    it "should set status" do
+      report.updated
+      expect(report.status).to eq "updated"
+    end
+  end
+
+  describe "#elapsed_time_in_milliseconds" do
+
+    context "when resource haven't finished" do
+      it "should return nil" do
+        expect(report.elapsed_time_in_milliseconds).to eq nil
+      end
+    end
+
+    context "when resource finished" do
+      it "should return the elapsed_time in milliseconds" do
+        report.finish
+        # zero because computers are fast
+        expect(report.elapsed_time_in_milliseconds).to eq 0
+      end
+    end
+  end
+
+  describe "#failed" do
+    let(:exception) { double("Chef::Exception::Test") }
+
+    it "should set exception and status" do
+      report.failed(exception)
+      expect(report.exception).to eq exception
+      expect(report.status).to eq "failed"
+    end
+  end
+
+  describe "#to_hash" do
+    context "for a simple_resource" do
+      let(:resource) do
+        z = Class.new(Chef::Resource) do
+          resource_name "zelda"
+        end
+        z.new("hyrule", run_context)
+      end
+      let(:hash) do
+        {
+          "after" => {},
+          "before" => {},
+          "delta" => "",
+          "duration" => "",
+          "id" => "hyrule",
+          "ignore_failure" => false,
+          "name" => "hyrule",
+          "result" => "create",
+          "status" => "unprocessed",
+          "type" => :zelda,
+        }
+      end
+
+      it "should work" do
+        expect(report.to_hash).to eq hash
+      end
+    end
+
+    context "for a lazy_resource that got skipped" do
+      let(:resource) do
+        c = Class.new(Chef::Resource) do
+          resource_name "link"
+          property :sword, String, name_property: true, identity: true
+        end
+        l = c.new("hyrule")
+        l.sword = Chef::DelayedEvaluator.new { nil }
+        l
+      end
+      let(:hash) do
+        {
+          "after" => {},
+          "before" => {},
+          "delta" => "",
+          "duration" => "",
+          "conditional" => "because",
+          "id" => "unknown identity (due to failed validation)",
+          "ignore_failure" => false,
+          "name" => "hyrule",
+          "result" => "create",
+          "status" => "skipped",
+          "type" => :link,
+        }
+      end
+      let(:conditional) do
+        double("Chef::Resource::Conditional", :to_text => "because")
+      end
+
+      before { report.skipped(conditional) }
```

Comment:
Since there's only one example, we can toss this in the `it` block like you've done with the other tests, yes?